### PR TITLE
Astro Prisma Accelerate Starter App

### PR DESCRIPTION
**Pull request by nurul3101**
This completed DA-871

### CODE REVIEW

File: accelerate/astro-starter/README.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Prisma Accelerate Example: Astro Starter
+
+This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](././src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/data-platform/accelerate/concepts#cache-strategies).
```
Comment (jharrell) [inline accelerate/astro-starter/README.md]:
```suggestion
This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/data-platform/accelerate/concepts#cache-strategies).
```

File: accelerate/astro-starter/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+# Accelerate connection string (used for queries by Prisma Client)
+DATABASE_URL="__YOUR_ACCELERATE_CONNECTION_STRING__"
+
+# Database connection string (used for migrations by Prisma Migrate)
+DIRECT_DATABASE_URL="__YOUR_DATABASE_CONNECTION_STRING__"
+
+```
+
+Note that `__YOUR_DATABASE_CONNECTION_STRING__` and `__YOUR_ACCELERATE_CONNECTION_STRING__` are placeholder values that you need to replace with the values of your database and Accelerate connection strings. Notice that the Accelerate connection string has the following structure: `prisma://accelerate.prisma-data.net/?api_key=__YOUR_ACCELERATE_API_KEY__`.
```
Comment (jharrell) [inline accelerate/astro-starter/README.md]: I don't think this info matches what is currently in the `.env.example`.

File: accelerate/astro-starter/src/env.d.ts
Comment (jharrell) [inline accelerate/astro-starter/src/env.d.ts:1]: Is this file needed?

File: accelerate/astro-starter/src/layouts/Layout.astro
Comment (jharrell) [inline accelerate/astro-starter/src/layouts/Layout.astro:1]: Could we switch from using tabs to spaces for indentation?

File: accelerate/astro-starter/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "@astrojs/react": "^3.6.0",
+    "@astrojs/tailwind": "^5.1.0",
+    "@types/react": "^18.3.3",
+    "@types/react-dom": "^18.3.0",
+    "astro": "^4.11.2",
+    "prisma": "^5.16.0",
+    "react": "^18.3.1",
+    "react-dom": "^18.3.1",
+    "tailwindcss": "^3.4.4",
+    "typescript": "^5.5.2",
```
Comment (jharrell) [inline accelerate/astro-starter/package.json]:
```suggestion
    "@astrojs/check": "0.7.0",
    "@astrojs/react": "3.6.0",
    "@astrojs/tailwind": "5.1.0",
    "@types/react": "18.3.3",
    "@types/react-dom": "18.3.0",
    "astro": "4.11.2",
    "prisma": "5.16.0",
    "react": "18.3.1",
    "react-dom": "18.3.1",
    "tailwindcss": "3.4.4",
    "typescript": "5.5.2",
```

File: accelerate/astro-starter/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider  = "postgresql"
+  url       = env("DATABASE_URL")
+  directUrl = env("DIRECT_URL")
+}
+
+model Quotes {
```
Comment (jharrell) [inline accelerate/astro-starter/prisma/schema.prisma]: We prefer models to be singular 
```suggestion
model Quote {
```

File: accelerate/astro-starter/prisma/migrations/20240627122402_init/migration.sql [hunk 5]
```diff
@@ -0,0 +1,8 @@
+-- CreateTable
+CREATE TABLE "Quotes" (
+    "id" SERIAL NOT NULL,
+    "quote" TEXT NOT NULL,
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    CONSTRAINT "Quotes_pkey" PRIMARY KEY ("id")
+);
```
Comment (jharrell) [inline accelerate/astro-starter/prisma/migrations/20240627122402_init/migration.sql]:
```suggestion
CREATE TABLE "Quote" (
    "id" SERIAL NOT NULL,
    "text" TEXT NOT NULL,
    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,

    CONSTRAINT "Quote_pkey" PRIMARY KEY ("id")
);
```

File: accelerate/astro-starter/prisma/schema.prisma [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  id        Int      @id @default(autoincrement())
+  quote     String
```
Comment (jharrell) [inline accelerate/astro-starter/prisma/schema.prisma]:
```suggestion
  text     String
```

File: accelerate/astro-starter/prisma/seed.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        quote:
+          'The future belongs to those who believe in the beauty of their dreams.',
+      },
+      {
+        quote:
+          'Do not wait for the perfect moment, take the moment and make it perfect.',
+      },
+      { quote: 'The only source of knowledge is experience.' },
+    ],
+  })
```
Comment (jharrell) [inline accelerate/astro-starter/prisma/seed.ts]:
```suggestion
  await prisma.quote.createMany({
    skipDuplicates: true,
    data: [
      { text: 'The only way to do great work is to love what you do.' },
      {
        text:
          'Success is not final, failure is not fatal: It is the courage to continue that counts.',
      },
      { text: 'In the middle of every difficulty lies opportunity.' },
      { text: "Believe you can and you're halfway there." },
      { text: 'The best way to predict the future is to create it.' },
      { text: "Don't watch the clock; do what it does. Keep going." },
      { text: 'The only thing we have to fear is fear itself.' },
      { text: 'The journey of a thousand miles begins with a single step.' },
      { text: 'If you can dream it, you can achieve it.' },
      { text: 'Innovation distinguishes between a leader and a follower.' },
      {
        text:
          'The greatest glory in living lies not in never falling, but in rising every time we fall.',
      },
      { text: "You miss 100% of the shots you don't take." },
      {
        text:
          'The only limit to our realization of tomorrow will be our doubts of today.',
      },
      { text: 'Change your thoughts and you change your world.' },
      {
        text:
          'To be yourself in a world that is constantly trying to make you something else is the greatest accomplishment.',
      },
      {
        text:
          "The only thing standing between you and your goal is the story you keep telling yourself as to why you can't achieve it.",
      },
      { text: 'Life is 10% what happens to us and 90% how we react to it.' },
      {
        text:
          'The future belongs to those who believe in the beauty of their dreams.',
      },
      {
        text:
          'Do not wait for the perfect moment, take the moment and make it perfect.',
      },
      { text: 'The only source of knowledge is experience.' },
    ],
  })
```

File: accelerate/astro-starter/src/components/Quote.astro [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    </h5>
+    <div class="spacer w-16"></div>
+    <p class="type text-xl font-medium text-blue-600 dark:text-blue-500">
+      {type}
+    </p>
+  </div>
+  <div class="content flex flex-col">
+    <p class="quote text-lg">
+      <span class="text-green-300">ID {data.id} </span>⸺ {'"'}
+      {data.quote}
```
Comment (jharrell) [inline accelerate/astro-starter/src/components/Quote.astro]:
```suggestion
      {data.text}
```

File: accelerate/astro-starter/src/types/quote.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+export interface QuoteData {
+  id: number
+  quote: string
+  createdAt: string
+}
```
Comment (jharrell) [inline accelerate/astro-starter/src/types/quote.ts]: We should maybe use the builtin Prisma type for this, but this also works.
```suggestion
export interface QuoteData {
  id: number
  text: string
  createdAt: string
}
```

**Review by jharrell — DISMISSED** [review]
Couple of changes needed, but looks really good.

File: accelerate/astro-starter/README.md (hunk 2 above)
Comment (nurul3101) [reply accelerate/astro-starter/README.md]: Made the changes ✅

File: accelerate/astro-starter/src/env.d.ts
Comment (nurul3101) [reply accelerate/astro-starter/src/env.d.ts:1]: Not needed. Removed.

File: accelerate/astro-starter/src/layouts/Layout.astro
Comment (nurul3101) [reply accelerate/astro-starter/src/layouts/Layout.astro:1]: Done.

File: accelerate/astro-starter/prisma/schema.prisma (hunk 4 above)
Comment (nurul3101) [reply accelerate/astro-starter/prisma/schema.prisma]: Done.

File: accelerate/astro-starter/prisma/migrations/20240627122402_init/migration.sql (hunk 5 above)
Comment (nurul3101) [reply accelerate/astro-starter/prisma/migrations/20240627122402_init/migration.sql]: Made the changes.

File: accelerate/astro-starter/package.json (hunk 3 above)
Comment (nurul3101) [reply accelerate/astro-starter/package.json]: Done.

File: accelerate/astro-starter/prisma/seed.ts (hunk 7 above)
Comment (nurul3101) [reply accelerate/astro-starter/prisma/seed.ts]: Done.

File: accelerate/astro-starter/src/types/quote.ts (hunk 9 above)
Comment (nurul3101) [reply accelerate/astro-starter/src/types/quote.ts]: Used these types for now, renamed quote to text.

File: accelerate/astro-starter/README.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
+# Prisma Accelerate Example: Astro Starter
+
+This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](./src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/data-platform/accelerate/concepts#cache-strategies).
+
+## Prerequisites
+
+To successfully run the project, you will need the following:
+
+- The **connection string** of a publicly accessible database.
+- Your **Accelerate connection string** (containing your **Accelerate API key**) which you can get by enabling Accelerate in a project in your [Prisma Data Platform](https://pris.ly/pdp) account (learn more in the [docs](https://www.prisma.io/docs/platform/concepts/environments#api-keys)).
```
Comment (ankur-arch) [inline accelerate/astro-starter/README.md]:
```suggestion
- Your **Accelerate connection string** (containing your **Accelerate API key**) which you can get by enabling Accelerate in a project in your [Prisma Data Platform](https://pris.ly/pdp) account (learn more in the [docs](https://www.prisma.io/docs/platform/about#api-keys)).
```

File: accelerate/astro-starter/README.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
+# Prisma Accelerate Example: Astro Starter
+
+This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](./src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/data-platform/accelerate/concepts#cache-strategies).
```
Comment (ankur-arch) [inline accelerate/astro-starter/README.md]:
```suggestion
This project showcases how to use Prisma ORM with Prisma Accelerate in a Astro application. It [demonstrates](./src/pages/route.ts#L13-16) every available [caching strategy in Accelerate](https://www.prisma.io/docs/accelerate/caching#cache-strategies).
```

**Review by ankur-arch — APPROVED** [review]

**Review by ankur-arch — APPROVED** [review]
